### DISCUSSION TITLE
How to forward a Stream to several Sinks with router strategy.

### DISCUSSION THREAD

**Discussion by lz1998** [tokio-rs/tokio#5807] [Q&A]
https://docs.rs/futures-util/0.3.28/futures_util/stream/trait.StreamExt.html#method.forward
```rust
fn route_choose_one<T:Sink,Item: Sink::Item>(item:Item, sinks:Vec<T>)-> T{
    // ...
}

let sinks = vec![sink1, sink2, sink3];
let stream = stream1;

stream.forward_with_strategy(sinks, route_choose_one)
```

Is there something to define a custom `Sink` like `stream!`, or wrap several `Sink`s in one?

**Comment by Darksonn** [comment]
There's no utility for this. You'll have to implement it yourself, e.g. with a loop like this:
```rs
while let Some(msg) = stream.next().await {
     sinks[pick_sink()].send(msg).await;
}
```
